x/sequencer/keeper/sequencer.go

```diff
@@ -171,3 +135,128 @@ func (k Keeper) removeUnbondingSequencer(ctx sdk.Context, sequencer types.Sequen
 	unbondingQueueKey := types.UnbondingSequencerKey(sequencer.SequencerAddress, sequencer.UnbondTime)
 	store.Delete(unbondingQueueKey)
 }
+
+/* -------------------------------------------------------------------------- */
+/*                                notice period                               */
+/* -------------------------------------------------------------------------- */
+// get finished notice period sequencers
+func (k Keeper) GetMatureNoticePeriodSequencers(ctx sdk.Context, endTime time.Time) (list []types.Sequencer) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := store.Iterator(types.NoticePeriodQueueKey, sdk.PrefixEndBytes(types.NoticePeriodQueueByTimeKey(endTime)))
+
+	defer iterator.Close() // nolint: errcheck
+
+	for ; iterator.Valid(); iterator.Next() {
+		var val types.Sequencer
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		list = append(list, val)
+	}
+
+	return
+}
+
+// set sequencer to notice period
+func (k Keeper) SetNoticePeriodQueue(ctx sdk.Context, sequencer types.Sequencer) {
+	store := ctx.KVStore(k.storeKey)
+	b := k.cdc.MustMarshal(&sequencer)
+
+	noticePeriodKey := types.NoticePeriodSequencerKey(sequencer.SequencerAddress, sequencer.UnbondTime)
+	store.Set(noticePeriodKey, b)
+}
+
+// remove sequencer from notice period queue
+func (k Keeper) removeNoticePeriodSequencer(ctx sdk.Context, sequencer types.Sequencer) {
+	store := ctx.KVStore(k.storeKey)
+	noticePeriodKey := types.NoticePeriodSequencerKey(sequencer.SequencerAddress, sequencer.UnbondTime)
+	store.Delete(noticePeriodKey)
+}
+
+/* ------------------------- proposer/next proposer ------------------------- */
+// GtAllProposers returns all proposers for all rollapps
+func (k Keeper) GetAllProposers(ctx sdk.Context) (list []types.Sequencer) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.ProposerByRollappKey(""))
+	iterator := sdk.KVStorePrefixIterator(store, []byte{})
```

Comment:
it's more readable to use ProposerKeyPrefix as the key here

Reply:
wanted to add the `KeySeparator` to the prefix as well

Reply:
ok ok, we can use this pattern